src/models/__tests__/service_preference.test.ts

```diff
@@ -57,6 +60,64 @@ const aRetrievedServicePreference: RetrievedServicePreference = {
   ...aStoredServicePreference
 };
 
+describe("ServicePreference::Codec", () => {
+  it("retrocompatibility - should succeed decoding a disabled inbox ServicePreference with enabled preferences and DENY accessReadMessageStatus", async () => {
+    const aWrongServicePreference = {
+      ...aStoredServicePreference,
+      isInboxEnabled: false,
+      accessReadMessageStatus: AccessReadMessageStatusEnum.DENY,
+      isEmailEnabled: true,
+      isWebhookEnabled: true
+    };
+
+    const result = ServicePreference.decode(aWrongServicePreference);
+
+    expect(E.isRight(result)).toBeTruthy();
+  });
+
+  it("should fail decoding a disabled inbox ServicePreference with enabled preferences and ALLOW accessReadMessageStatus", async () => {
+    const aWrongServicePreference = {
+      ...aStoredServicePreference,
+      isInboxEnabled: false,
+      accessReadMessageStatus: AccessReadMessageStatusEnum.ALLOW,
+      isEmailEnabled: true,
+      isWebhookEnabled: true
+    };
+
+    const result = ServicePreference.decode(aWrongServicePreference);
+
+    expect(E.isLeft(result)).toBeTruthy();
+  });
+
+  it("should succeed decoding a correctly disabled ServicePreference", async () => {
+    const aWrongServicePreference = {
```

Comment:
ditto

Reply:
fixed in https://github.com/pagopa/io-functions-commons/pull/270/commits/2c5fe322763ae40ed11b6e23fe10a60620fb2835